subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -551,41 +694,50 @@ static void http_fota_handler(const struct fota_download_evt *evt)
 
 	switch (evt->id) {
 	case FOTA_DOWNLOAD_EVT_FINISHED:
-		LOG_DBG("FOTA download completed evt received");
+		LOG_DBG("FOTA_DOWNLOAD_EVT_FINISHED");
 
 		/* Always send download complete progress */
 		aws_fota_evt.id = AWS_FOTA_EVT_DL_PROGRESS;
 		aws_fota_evt.dl.progress = AWS_FOTA_EVT_DL_COMPLETE_VAL;
 		callback(&aws_fota_evt);
 
-		fota_state = APPLY_UPDATE;
-		err = update_job_execution(c, job_id, AWS_JOBS_IN_PROGRESS, "");
-		if (err != 0) {
+		err = update_job_execution(c, job_id_handling, AWS_JOBS_SUCCEEDED, "");
+		if (err != 0 && connected) {
 			aws_fota_evt.id = AWS_FOTA_EVT_ERROR;
 			callback(&aws_fota_evt);
+			state_clear();
+			return;
 		}
+
+		state_set(STATE_DOWNLOAD_COMPLETE);
 		break;
 
 	case FOTA_DOWNLOAD_EVT_ERASE_PENDING:
+		LOG_DBG("FOTA_DOWNLOAD_EVT_ERASE_PENDING");
 		aws_fota_evt.id = AWS_FOTA_EVT_ERASE_PENDING;
 		callback(&aws_fota_evt);
 		break;
 
 	case FOTA_DOWNLOAD_EVT_ERASE_DONE:
+		LOG_DBG("FOTA_DOWNLOAD_EVT_ERASE_DONE");
 		aws_fota_evt.id = AWS_FOTA_EVT_ERASE_DONE;
 		callback(&aws_fota_evt);
 		break;
 
 	case FOTA_DOWNLOAD_EVT_ERROR:
-		LOG_ERR("FOTA download failed, report back");
-		fota_state = NONE;
-		execution_state = AWS_JOBS_QUEUED;
-		(void)update_job_execution(c, job_id, AWS_JOBS_FAILED, "");
+		LOG_ERR("FOTA_DOWNLOAD_EVT_ERROR");
+		(void)update_job_execution(c, job_id_handling, AWS_JOBS_FAILED, "");
 		aws_fota_evt.id = AWS_FOTA_EVT_ERROR;
 		callback(&aws_fota_evt);
+		state_clear();
+		/* If the FOTA download fails it might be due to the image being deleted.
+		 * Try to get the next job if any exist.
+		 */
+		(void)aws_jobs_get_job_execution(c, "$next", get_topic);
```

Comment:
We need some blank lines in this block to separate function calls, variable assignment and comments